src/Rules/Keywords/RequireFileExistsRule.php

```diff
@@ -0,0 +1,130 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Keywords;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\BinaryOp\Concat;
+use PhpParser\Node\Expr\ClassConstFetch;
+use PhpParser\Node\Expr\ConstFetch;
+use PhpParser\Node\Expr\Include_;
+use PhpParser\Node\Identifier;
+use PhpParser\Node\Name;
+use PhpParser\Node\Scalar\MagicConst\Dir;
+use PhpParser\Node\Scalar\String_;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\ReflectionProvider;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleErrorBuilder;
+use function constant;
+use function defined;
+use function dirname;
+use function file_exists;
+use function is_string;
+use function sprintf;
+
+/**
+ * @implements Rule<Include_>
+ */
+final class RequireFileExistsRule implements Rule
+{
+
+	public function __construct(private ReflectionProvider $reflectionProvider)
+	{
+	}
+
+	public function getNodeType(): string
+	{
+		return Include_::class;
+	}
+
+	public function processNode(Node $node, Scope $scope): array
+	{
+		if ($this->shouldProcessNode($node)) {
+			$filePath = $this->resolveFilePath($node->expr, $scope);
+			if ($filePath !== null && !file_exists($filePath)) {
+				return [
+					RuleErrorBuilder::message(
+						sprintf(
+							'Required file "%s" does not exist.',
+							$filePath,
+						),
+					)->build(),
+				];
+			}
+		}
+
+		return [];
+	}
+
+	private function shouldProcessNode(Node $node): bool
+	{
+		return $node instanceof Include_ && (
```

Comment:
But there isn't any validation that the `$node` would be a `Include_` except for trusting the PHPStan itself, right? Otherwise, phpstan would be also whining about it.

Reply:
We can trust PHPStan. Generics take care of that.